connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java

```diff
@@ -89,8 +91,22 @@ private ResponseEntity<?> processWebhook(
       ActiveInboundConnector connector, WebhookProcessingPayload payload) {
     ResponseEntity<?> connectorResponse;
     try {
-      var webhookResult =
-          ((WebhookConnectorExecutable) connector.executable()).triggerWebhook(payload);
+      WebhookConnectorExecutable connectorHook =
+          (WebhookConnectorExecutable) connector.executable();
+
+      if (connectorHook.verificationExpression() != null) {
+        var verificationResult = connectorHook.verify(payload);
+        if (verificationResult != null) {
+          HttpHeaders headers = new HttpHeaders();
+          Optional.of(verificationResult)
```

Comment:
`verificationResult` should never be null here.

Reply:
Why not? If it's `null` (validation failed), then return immediately - fail fast strategy. Otherwise, we need to introduce some sort of flag whether vaildation result was "success" or "failure". WDYT?

Reply:
Because you already checked it with: ```verificationResult != null```

Reply:
Wondering why the use of Optional here.

Reply:
aaahhh, I see. Sorry, got confused myself :)
Yes, some of those will be cleaned up. As a part of our commitment, we have to show progress every two days, so the code you seeing is still in progress.

Reply:
`Optional` still makes sense here, since I am using `map` and `ifPresent`-`forEach` structure. Without `Optional` I would need to write a `for`-loop.